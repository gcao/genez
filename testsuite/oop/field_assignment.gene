#!/usr/bin/env gene run

# Test: Direct assignment to instance variables (= /name value)

(class Counter
  (var /count)
  
  (.ctor []
    (= /count 0))
  
  (.fn increment []
    (= /count (+ /count 1)))
  
  (.fn set_count [n]
    (= /count n)))

(class Person
  (var /first_name)
  (var /last_name)
  
  (.fn set_name [first last]
    (do
      (= /first_name first)
      (= /last_name last)))
  
  (.fn update_first_name [name]
    (= /first_name name)))

# Complex assignments
(class Calculator
  (.prop result)
  (.prop memory)
  
  (.fn calculate [op a b]
    (do
      (= /result 
        (if (== op "+") (+ a b)
        else (if (== op "-") (- a b)
        else (if (== op "*") (* a b)
        else (if (== op "/") (/ a b)
        else 0)))))
      (= /memory /result))))

(print "Field assignment syntax parsed successfully")