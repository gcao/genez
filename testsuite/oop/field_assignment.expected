Field assignment syntax parsed successfully